pandas/_typing.py

```diff
@@ -84,6 +84,7 @@
 FrameOrSeries = TypeVar("FrameOrSeries", bound="NDFrame")
 
 Axis = Union[str, int]
+Frequency = Optional[str]
```

Comment:
this also includes DateOffset

Reply:
maybe better to also remove the optional here, xref https://github.com/pandas-dev/pandas/pull/38416/files#r552673889